src/repr/src/row.rs

```diff
@@ -1583,6 +1841,14 @@ impl RowRef {
         }
     }
 
+    /// Iterator the `DatumKind` elements of the `Row`.
+    pub fn kind_iter(&self) -> DatumKindIter {
```

Comment:
Nit: I feel like `iter_kinds` or `kinds` would be more in line with how Rust names similar methods on containers in the stdlib.

```suggestion
    pub fn iter_kinds(&self) -> DatumKindIter {
```